libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -127,6 +119,18 @@ string IRGenerator::generate(Block const& _block)
 	return generator.code();
 }
 
+void IRGenerator::collectFunctions(ContractDefinition const& _contract)
+{
+	for (auto const* contract: _contract.annotation().linearizedBaseContracts)
+		for (auto const* fun: contract->definedFunctions())
+			if (fun->isVisibleViaContractTypeAccess())
+				m_context.functionGenerationQueue().push(fun);
```

Comment:
We might first check if the function has already been generated.

Reply:
I thought it would be superfluous since `m_context.functionCollector()` ignores functions it has already collected.

Chcecking it here would save us some inserts into a `set` and some insignificant amount of memory but it seemed like an unnecessary micro-optimization to me.

Reply:
Done (after further discussion in https://github.com/ethereum/solidity/pull/8663#issuecomment-614586780).